src/main/java/org/openml/webapplication/features/GlobalMetafeatures.java

```diff
@@ -23,36 +23,59 @@
     private int expectedQualities;
     private ArrayList<String> expectedIds = new ArrayList<>();
 
-    public GlobalMetafeatures(Integer window_size) throws Exception {
-        Characterizer[] characterizers = {
-                new SimpleMetaFeatures(), // done before, but necessary for streams
-                new Statistical(),
-                new NominalAttDistinctValues(),
-                new AttributeEntropy(),
-                new GenericLandmarker("kNN1N", cp1NN, 2, null),
-                new GenericLandmarker("NaiveBayes", cpNB, 2, null),
-                new GenericLandmarker("DecisionStump", cpDS, 2, null),
-                new GenericLandmarker("CfsSubsetEval_kNN1N", cpASC, 2, Utils.splitOptions(preprocessingPrefix + cp1NN)),
-                new GenericLandmarker("CfsSubsetEval_NaiveBayes", cpASC, 2, Utils.splitOptions(preprocessingPrefix + cpNB)),
-                new GenericLandmarker("CfsSubsetEval_DecisionStump", cpASC, 2, Utils.splitOptions(preprocessingPrefix + cpDS))
-        };
-        batchCharacterizers = new ArrayList<>(Arrays.asList(characterizers));
-        // additional parameterized batch landmarkers
-        String zeros = "0";
-        for( int i = 1; i <= 3; ++i ) {
-            zeros += "0";
-            String[] j48Option = { "-C", "." + zeros + "1" };
-            batchCharacterizers.add(new GenericLandmarker("J48." + zeros + "1.", "weka.classifiers.trees.J48", 2, j48Option));
+    /**
+     * Collection of characterizers
+     * 
+     * @param characterizer Class name of requested characterizer. Can also be 'fast' 
```

Comment:
I am personally not a big fan of adding string options with a magical meaning (i.e., fast and generic landmarker). This will in the future abolish us from making an automatic mappings. All characterizers that are ran are uniquely identifiable with a string name. Why not take advantage of that and pass this unique string name to the function. This seems like a much more understandable and maintainable design pattern, which also removes the additional if/else structure that was created

Reply:
I only added 'fast' because I thought that might be handy. Then you can recalculate all fast metafeatures with a single command. If you don't agree I will remove it to simplify the code.